packages/zoe/src/contractSupport/recorder.js

```diff
@@ -252,7 +252,7 @@ export const prepareMockRecorderKitMakers = () => {
  * explictly specify the type that the Pattern will verify through a match.
  *
  * @template T
- * @typedef {Matcher & {validatedType?: T}} TypedMatcher
+ * @typedef {Pattern & {validatedType?: T}} TypedMatcher
```

Comment:
The original should always have been `Pattern` rather than `Matcher`. Good that hiding the `Matcher` type reveals these. For most purposes, the `Matcher` type is a detail within the meaning of the `Pattern` type.